### PR TITLE
Feature/stub out methods

### PR DESCRIPTION
**Pull request by CampbellJGilpin**

### CODE REVIEW

File: v2/AdminCore.Common/Interfaces/ITeamService.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+using AdminCore.DTOs.Team;
+using System.Collections.Generic;
+
+namespace AdminCore.Common.Interfaces
+{
+    public interface ITeamService
+    {
+        IList<TeamDto> findAll();
```
Comment (BrownMark) [inline v2/AdminCore.Common/Interfaces/ITeamService.cs]: Formatting inconsistent

File: v2/AdminCore.DTOs/Team/TeamDto.cs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+﻿namespace AdminCore.DTOs.Team
+{
+    public class TeamDto
+    {
+        public int ClientId { get; set; }
```
Comment (BrownMark) [inline v2/AdminCore.DTOs/Team/TeamDto.cs]: Again on formatting.

File: v2/AdminCore.Services/Mappings/ServiceMapperProfile.cs [hunk 3]
```diff
@@ -7,30 +7,37 @@
...
-  /// The service mapper profile.
-  /// </summary>
-  public class ServiceMapperProfile : Profile
-  {
     /// <summary>
-    /// Initializes a new instance of the <see cref="ServiceMapperProfile"/> class.
+    /// The service mapper profile.
     /// </summary>
-    public ServiceMapperProfile()
+    [Obsolete("Not used anymore - Being replaced by individual mapping classes", true)]
```
Comment (BrownMark) [inline v2/AdminCore.Services/Mappings/ServiceMapperProfile.cs]: nice

File: v2/AdminCore.WebApp/Controllers/TeamController.cs [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+        [HttpPost("/")]
+        [AllowAnonymous]
+        public void createTeam([FromBody] TeamViewModel teamViewModel)
+        {
+            TeamDto teamDto = _mapper.Map<TeamDto>(teamViewModel);
+            _teamService.save(teamDto);
+        }
+
+        [HttpPut("/")]
```
Comment (BrownMark) [inline v2/AdminCore.WebApp/Controllers/TeamController.cs]: Don't needs routes for these.

File: v2/AdminCore.WebApp/Models/Team/TeamViewModel.cs [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace AdminCore.WebApi.Models.Team
+{
+    public class TeamViewModel
+    {
+        private int clientId;
+        private int teamId;
```
Comment (BrownMark) [inline v2/AdminCore.WebApp/Models/Team/TeamViewModel.cs]: Formatting.

**Review by BrownMark — CHANGES_REQUESTED** [review]

File: v2/AdminCore.Common/Interfaces/ITeamService.cs (hunk 1 above)
Comment (CampbellJGilpin) [reply v2/AdminCore.Common/Interfaces/ITeamService.cs]: Method name, or also indenting?

File: v2/AdminCore.DTOs/Team/TeamDto.cs (hunk 2 above)
Comment (CampbellJGilpin) [reply v2/AdminCore.DTOs/Team/TeamDto.cs]: Indenting?

File: v2/AdminCore.Common/Interfaces/IEventService.cs [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+  {
+    IList<EventDto> FindAnnualLeaveByEmployee(int EmployeeID);
+
+    IList<EventDto> FindByDateBetween(DateTime RangeStart, DateTime RangeEnd, EventTypes EventType);
+
+    IList<EventDto> FindByEmployee(int EmployeeID);
+
+    EventDto FindByEmployeeIDStartDateAndEndDate(int EmployeeID, DateTime StartDate, DateTime EndDate);
+
+    EventDto FindById(int ID);
```
Comment (BrownMark) [inline v2/AdminCore.Common/Interfaces/IEventService.cs]: convention is camel case for variables id or employeeId

File: v2/AdminCore.Services/EventService.cs [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+    {
+      throw new System.NotImplementedException();
+    }
+
+    public void SaveEvents(List<EventDto> EventDtos)
+    {
+      throw new System.NotImplementedException();
+    }
+
+    public void UpdateEvents(UpdateEventDto UpdateEventDto)
```
Comment (BrownMark) [inline v2/AdminCore.Services/EventService.cs]: Same again with conventions

File: v2/AdminCore.WebApp/Controllers/ClientController.cs [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+    private readonly IClientService _clientService;
+
+    public ClientController(IClientService clientService, IMapper mapper)
+    {
+      _clientService = clientService;
+      _mapper = mapper;
+    }
+
+    [HttpGet]
+    [AllowAnonymous]
```
Comment (BrownMark) [inline v2/AdminCore.WebApp/Controllers/ClientController.cs]: Can we remove AllowAnnoymous everywhere? Just remove the Authorise

File: v2/AdminCore.WebApp/Controllers/TeamController.cs [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+        private ITeamService _teamService;
+
+        public TeamController(ITeamService teamService, IMapper mapper)
+        {
+            _teamService = teamService;
+            _mapper = mapper;
+        }
+
+        [HttpGet("/{clientId}")]
+        [AllowAnonymous]
```
Comment (BrownMark) [inline v2/AdminCore.WebApp/Controllers/TeamController.cs]: Same here

**Review by BrownMark — CHANGES_REQUESTED** [review]

File: v2/AdminCore.Services/ClientService.cs [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+namespace AdminCore.Services
+{
+  public class ClientService : IClientService
+  {
+    public IList<ClientDto> GetAll()
+    {
+      throw new System.NotImplementedException();
+    }
+
+    public void UpdateClient(ClientDto missing_name)
```
Comment (BrownMark) [inline v2/AdminCore.Services/ClientService.cs]: Can we name this client?

File: v2/AdminCore.Common/Interfaces/IClientService.cs [hunk 11]
```diff
@@ -0,0 +1,12 @@
+using System.Collections.Generic;
+using AdminCore.DTOs.Client;
+
+namespace AdminCore.Common.Interfaces
+{
+  public interface IClientService
+  {
+    IList<ClientDto> GetAll();
+
+    void UpdateClient(ClientDto);
```
Comment (BrownMark) [inline v2/AdminCore.Common/Interfaces/IClientService.cs]: Does this compile?

**Review by BrownMark — CHANGES_REQUESTED** [review]

File: v2/AdminCore.Services/EventService.cs (hunk 7 above)
Comment (CampbellJGilpin) [reply v2/AdminCore.Services/EventService.cs]: This file has been overwritten and no longer exists in branch

File: v2/AdminCore.Common/Interfaces/IEventService.cs (hunk 6 above)
Comment (CampbellJGilpin) [reply v2/AdminCore.Common/Interfaces/IEventService.cs]: This file has been overwritten and no longer exists in branch

**Comment by CampbellJGilpin** [comment]
Addressed most recent reviews - and resolved conflicts

File: v2/AdminCore.WebApp/Controllers/TeamController.cs [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+        }
+
+        [HttpPut]
+        public void saveTeam([FromBody] TeamViewModel teamViewModel)
+        {
+            TeamDto teamDto = _mapper.Map<TeamDto>(teamViewModel);
+            _teamService.Save(teamDto);
+        }
+
+        private List<TeamViewModel> mapTeamsToDtos(List<TeamDto> teams)
```
Comment (BrownMark) [inline v2/AdminCore.WebApp/Controllers/TeamController.cs]: You don't need this method

File: v2/AdminCore.WebApp/Controllers/TeamController.cs [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+        public TeamController(ITeamService teamService, IMapper mapper)
+        {
+            _teamService = teamService;
+            _mapper = mapper;
+        }
+
+        [HttpGet("/{clientId}")]
+        public List<TeamViewModel> findAllTeamsForClientId(int clientId)
+        {
+            return mapTeamsToDtos(_teamService.FindByClient(clientId));
```
Comment (BrownMark) [inline v2/AdminCore.WebApp/Controllers/TeamController.cs]: this should be _mapper.Map<IList<TeamDto>, List<TeamViewModel>>(teams)

**Review by BrownMark — CHANGES_REQUESTED** [review]

**Comment by eoinmcafee** [comment]
Hey @CampbellJGilpin -  What's happening with this PR? Looks like there's conflicts and needs develop merged.

**Comment by CampbellJGilpin** [comment]
@eoinmcafee @BrownMark 

Sorry, hadn't realized. Thanks for merging develop.

The two outstanding requests are addressed for TeamController

**Review by eoinmcafee — APPROVED** [review]

**Review by BrownMark — APPROVED** [review]

**Pull request merged**